docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,77 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the implementation.
+The implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to implementation- or application-requested form.
+
+The attribute value `normalization="disabled"` can be used to indicate that no automatic normalization happens.
+Using this setting is not recommended. See [`<settings>`](#element-settings) for details.
+
+Input source files may be in any normalization format, however, authors should be aware of two areas where normalization affects keyboard operation: that of transform matching, and that of output.
+
+Normalization can be applied within segments defined by Grapheme Cluster Boundaries as defined by [UAX #29](https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries). This segmentation is pertinent where markers are involved, see below.
+
+### Normalization and Transform Matching
+
+Regardless of the normalization form in the keyboard source file or in the edit buffer context, transform matching will be performed using **NFD**. For example, all of the following transforms will match the input strings `è̠`, whether the input is U+00E8 U+0320, U+0065 U+0320 U+0300, or U+0065 U+0300 U+0320.
+
+```xml
+<transform from="e\u{0300}\u{0320}" /> <!-- Non-NFD but will match -->
+<transform from="e\u{0320}\u{0300}" />
+<transform from="\u{00E8}\u{0320}" /> <!-- è + U+0320 -->
+```
+
+#### Normalization and Markers
+
+A special issue occurs when markers are involved. The markers may be reordered along with characters.
+
+**Example 1**
+
+Consider this example, without markers:
+
+- `e\u{0300}\u{0320}` (original)
+- `e\u{0320}\u{0300}` (NFD)
+
+If we add markers:
+
+- `e\u{0300}\m{marker}\u{0320}` (original)
+- `e\m{marker}\u{0320}\u{0300}` (NFD)
+
+The principle is that the marker stays 'glued' to the following character, in this case the `\u{0320}`. If a marker occurs at the end of input or at the end of a segment (a grapheme cluster boundary), the marker is 'glued' to the end of that segment during the course of normalization. As another example:
```

Comment:
Noted during the meeting:  give rationale to attaching the marker to the following character.

Association of an "out of band" element interstitched with actual context appears to be novel and may be of use to others outside of LDML, so we should probably elaborate / motivate our decisions here more strongly.

Reply:
Possible aspect to this:

- marker before character:  this is a special property of the character just typed
- character before marker:  no matter where the character ends up, activate a special state that affects the next keystroke